rotkehlchen/db/snapshots.py

```diff
@@ -82,6 +85,31 @@ def get_timed_location_data(
             )
         return location_data
 
+    def get_main_currency_price(self, timestamp: Timestamp) -> Tuple[Asset, Price]:
+        """Gets the main currency and its equivalent price at a particular timestamp."""
+        main_currency = self.db.get_main_currency()
+        main_currency_price = None
+        try:
+            main_currency_price = PriceHistorian.query_historical_price(
+                from_asset=A_USD,
+                to_asset=main_currency,
+                timestamp=timestamp,
+            )
+        except NoPriceForGivenTimestamp:
+            main_currency_price = Price(ONE)
+            MessagesAggregator().add_error(
```

Comment:
Hmm no this will not work Isaac. `MessagesAggregator` is not a singleton class. The messages aggregator, is a normal class that is instantiated when the `Rotkehlchen` object is and is passed around.

What you are effectively doing here is creating a new MessagesAggregator, adds an error, and then it just stays in memory forever until it is garbage collected. The error will never be seen by the frontend.

You need to pass the MessagesAggregator object as an argument to this function.